### PR TITLE
redesign request notification dialog

### LINKED ISSUE #2085: Implement redesign of raised hand notification dialog
To further clarify some design details that aren't self explanatory by the exported design:

- Width of the notifications should be defined by the length of the usernames

### LINKED ISSUE #2192: Implement redesign of join request dialog

### PR DESCRIPTION
**Pull request by Schwehn42**
## Description

[//]: <> (Explain the changes you’ve made. It doesn’t need to be fancy and you don’t have to get too technical.)
**This PR implements new designs for both Join -and RaiseHand request notifications.**
- resolves #2085 
- resolves #2192 

## Changelog

[//]: <> (Please provide a detailed list of the changes you have made to the codebase.)
- refactored the Request component by splitting it into two files, `Requests.tsx` and `Request.tsx`
- former is a container for all requests, latter is the request itself.
- implemented design and transistions according to design

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] The light- and dark-theme are both supported and tested
- [x] The design was implemented and is responsive for all devices and screen sizes
- [x] The application was tested in the most commonly used browsers (e.g. Chrome, Firefox, Safari)

## (Optional) Visual Changes

[//]: <> (If available, please provide a before and after screenshot of your UI related changes.)

https://user-images.githubusercontent.com/7889564/196410330-8a7995ac-7100-493f-ae2f-2708dd7adb56.mov

### CODE REVIEW

File: src/components/Requests/Request/Request.scss
Comment (CronJorian) [inline src/components/Requests/Request/Request.scss:29]: Why padding and margin?
Wouldn't margin be enough? And could you use $padding-default or $padding-small?

File: src/components/Requests/Request/Request.scss
Comment (CronJorian) [inline src/components/Requests/Request/Request.scss:88]:
```suggestion
  border-radius: $border-radius--round;
```

File: src/components/Requests/Request/Request.tsx
Comment (CronJorian) [inline src/components/Requests/Request/Request.tsx:9]: no need for exporting, when only used in `Request.tsx`

```suggestion
type RequestProps = {
```

File: src/components/Requests/Request/Request.tsx
Comment (CronJorian) [inline src/components/Requests/Request/Request.tsx:44]: Could not find `request__text-container`. Is this comment still relevant?

File: src/components/Requests/Request/Request.tsx
Comment (CronJorian) [inline src/components/Requests/Request/Request.tsx:45]: imho there is no need to call it "middle-wrapper" instead of "wrapper".

File: src/components/Requests/Requests.tsx
Comment (CronJorian) [inline src/components/Requests/Requests.tsx:28]: should also work as: `return requests.length || participantsWithRaisedHand.length ? (` because 0 is a falsy value

**Review by CronJorian — COMMENTED** [review]
All in all well done, just some small nitpicks. :)

File: src/components/Requests/Request/Request.scss
Comment (CronJorian) [reply src/components/Requests/Request/Request.scss]: Not sure whether I get what you mean. The :hover pseudo class had only the two right spacings and I just wondered why you need to push the outer _and_ inner box instead of just the outer (`margin-right: 20px;`) or the inner (`padding-right: 20px;`). That was why I asked :)
The effect per se is nice :)

File: src/components/Requests/Requests.scss
Comment (CronJorian) [reply src/components/Requests/Requests.scss:9]: Would be an option, you can check if theres an option to add an shadow if the list is scrollable (e.g. https://css-scroll-shadows.vercel.app/ resp. https://lea.verou.me/2012/04/background-attachment-local/) so the scrollable is clear.

Though more easily is including the scrollbar mixin (`@include scrollbar();`) (and maybe changing the color) The scrollbar is only shown if theres scrollable content :)

**Review by CronJorian — CHANGES_REQUESTED** [review]
Sorry for the late review, there are still some things I noticed:
- The request window is blocking the elements behind:

https://user-images.githubusercontent.com/44020029/201945655-18e53332-d33d-48e1-9398-34c388af2596.mov

- Not sure if the div should always be the full height, but I guess its fine as long it doesn't block elements below from being accessed
  - you can use `pointer-events: none;` on `requests__wrapper` and `pointer-events: all;` on `requests__main` for a quickfix, but i am not sure if this is a bad practice or not.

- The request is above the settings portal which leads to ugly moments

image

- The box shadow to the left and right is cut off due to the width of the parent element

image

One time after processing a join request the backend crashed, but I am not sure whether it's because of any of these changes, just wanted to let you know.
These were the issues I found, I think after addressing these, we'll just merge it and change other stuff later on (if there is more to change)

Oh and I think while having 10-20 requests at a time the browser slows down while hovering. This might be linked to the cut of sides of the box shadow. Would be nice if you could test this too (just open multiple private windows in safari) and check the performance.

File: src/components/Requests/Requests.scss
Comment (Schwehn42) [reply src/components/Requests/Requests.scss:9]: i've made some further adjustments, the requests container is now no longer full height. 

Also, I'm not certain if you would like the scrollbar to be shown when hovering (and the list is scrollable) or don't show it at all times?

**Comment by Schwehn42** [comment]
@CronJorian shadows should be fixed now.  
Not sure what caused the backend crash, I'm just gonna claim it didn't happen because of any of my changes 👀

File: src/constants/style.scss
Comment (CronJorian) [inline src/constants/style.scss:17]: Is this comment necessary? :)

**Comment by CronJorian** [comment]
Looks really good now :)

Is it intended that the request container has just a light theme? If so, i'll approve, if not can you add these changes and then we're done here 😅

File: src/constants/style.scss
Comment (scja1069) [reply src/constants/style.scss]: nope ^^

**Review by Schwehn42 — COMMENTED** [review]
@CronJorian I think we should be good to go now.  
At most one could add a second icon for the notification type "Join", but I couldn't find a fitting one in our exisiting assets, maybe we can add that later?

**Review by CronJorian — APPROVED** [review]

**Pull request merged**
